### ISSUE TITLE
Why can't the compiler create executables

### ISSUE THREAD

**Issue by hmaarrfk** [Archiconda/build-tools#3]
@jjhelmus Thanks for all your help bootstrapping.

Do you know why I'm unable to create executables with the compilers you provided:

```
checking whether the C compiler works... no
configure: error: in `/root/archiconda3/conda-bld/make_1546046933373/work':
configure: error: C compiler cannot create executables
See `config.log' for more details
```
https://app.shippable.com/github/Archiconda/make-make-feedstock/runs/1/1/console

I had to rename the repository `make-make-feedstock` because I deleted it by accident and shippable freaked out that a new repo was created with the same "old name" as before.

**Comment by jjhelmus** [comment]
My guess is that the failure is caused by the older glibc version in the container.  The compilers in the boostrap channel require glibc 2.27 (Ubuntu 18.04) but the container provides glib 2.23 (Ubuntu 16.04).  If you can figure out how to dump the `config.log` file the error message should provide additional details.

**Comment by hmaarrfk** [comment]
is that it...... man i must have read 16.04 accidentally.......

**Comment by jjhelmus** [comment]
I might have claimed the compilers were built on Ubunutu 16.04 at some point.  I was on vacation the second half of December and did not have access to my aarch64 development box so wasn't certain on what I had installed.  I checked today and can confirm that is was Ubuntu 18.04 not 16.04.

**Comment by hmaarrfk** [comment]
also, why can't raspbian update to aarch64 already.....

man this is dumb, make needs make to compile...

https://app.shippable.com/github/Archiconda/make-make-feedstock/runs/3/1/console

**Comment by hmaarrfk** [comment]
(but yes, you are right, the compilers you mentioned do need 18.04)

**Comment by jjhelmus** [comment]
Raspbian is compatible with all Raspberry Pi models so it cannot change to aarch64 as the Raspberry Pi 1 and the Zero are armv6l CPUs.

**Comment by hmaarrfk** [comment]
I totally understand that, but it is real shame since this kinda means we would have to support two architechtures....

i'm not sure shippable gives you real aarch32, but maybe just an image that starts up as 32 bit OS.
this was causing problems with CPU detection AFAIK

**Comment by hmaarrfk** [comment]
Nice, well the shippable ci is now mostly confirmed working.

this was uploaded!!!!
https://anaconda.org/archiarm/make

**Comment by jjhelmus** [comment]
There are a lot of armv7 boards available besides the Raspberry Pi that would benefit from aarch32 packages.  Most Linux distributions support two different ARM platforms, aarch64 and armhf.  armhf is for 32-bit armv7l machines with hardware floating point.

conda determines the platform from `platform.machine` which I believe will match the output of the `uname -p` command.  My guess is that shippable's aarch32 platform is running on 64-bit hardware so `uname -a` probably is reporting aarch64.  There are some methods to modify the architecture reported by uname (e.g. on x86 the `linux32` command).  Setting the environment variable `CONDA_SUBDIR=linux-armv7l` may be sufficient for the case here.

**Comment by jjhelmus** [comment]
> Nice, well the shippable ci is now mostly confirmed working.

Neat!

**Comment by hmaarrfk** [comment]
got it. well i'll wait for your compilers now. even if it is just for 64 bit.

I don't really know how you bootstrapped things, but Archiconda 0.1.1 was able to run on ubuntu 16.04, so if you want, we can use the scripts here to recompile everything with the compiler you are working on.

**Comment by hmaarrfk** [comment]
I think the compilers in c4aarch64 are tuned to work much better than the ones in bootstrap.

I'm going to have to figure out channel priorities to create a new installer that sets up c4aarch64 and archiarm as higher priority channels

**Comment by jjhelmus** [comment]
> I think it is because some scripts assume that HOST and BUILD should be the same, or very similar, or very dissimily. Not almost the same

Yes `BUILD` needs to be set explicitly in `conda_build_config.yaml` to `"aarch64-conda_cos7-linux-gnu"`.  If this if not done [conda build](https://github.com/conda/conda-build/blob/432fd21b4202706d737b92b86c269e9086363ee5/conda_build/environ.py#L580-L610) will set the BUILD variable to `aarch64-conda_cos6-linux-gnu`.  I added the `conda_build_config.yaml` file I've been using to the [conda4aarch64 repository](https://github.com/jjhelmus/conda4aarch64/blob/master/recipes/conda_build_config.yaml).

**Comment by hmaarrfk** [comment]
Yeah, I started to create a default .ci_setup file that introduced much of that. I guess there is no way around this.

https://github.com/Archiconda/conda-smithy/commit/a5d290b93e5fac234901114bd780225baa0d0aee
I'll have to add a few more of the parameters you have listed in there, but I think for the most part that was the most important one.

I guess you aren't working much with the conda-forge infrastrucutre.

Is there a config file we should change in the installer to make this the default?

**Comment by jjhelmus** [comment]
> Is there a config file we should change in the installer to make this the default?

I do not think the installer can set this but the default should be changed in conda-build.
